src/backend/t/lib/Test/Mock/BSRPC.pm

```diff
@@ -57,7 +57,12 @@ BEGIN {
 
       my $receiver = $param->{'receiver'};
       if ($receiver) {
-	$ret = $receiver->(BSHTTP::str2req($ret), $param, $xmlargs || $param->{'receiverarg'}) if $receiver;
+        if ($receiver == \&BSHTTP::file_receiver) {
+          $ret = $receiver->(BSHTTP::str2req($ret), $param, $xmlargs || $param->{'receiverarg'});
+        }
+        elsif ($receiver == \&BSHTTP::cpio_receiver) {
+          $ret = $receiver->({ '__data' => $ret }, $param, $xmlargs || $param->{'receiverarg'});
+        }
```

Comment:
I don't get why this is needed. BSHTTP::str2req does exactly what you do here, so why doesn't the original code work?

Reply:
True. :man_facepalming: I reverted this change.